tests/python_version_test.rs

```diff
@@ -175,7 +175,7 @@ fn python_version_file_invalid_version() {
                 isn't in the correct format.
                 
                 The following version was found:
-                3.12.0invalid
+                3.12.0�
                 
                 However, the Python version must be specified as either:
                 1. The major version only, for example: {DEFAULT_PYTHON_VERSION} (recommended)
```

Comment:
@edmorley it might be worth describing in this error message what it means when `�` shows up in the incorrectly formatted version reported. E.g.; 
> 3. If the version reported above contains `�` characters, these will have to be removed. Most IDEs can be configured to display hidden characters which can help in locating these.

Reply:
For the error message, there are a lot of things we have to tell the user (since there are a lot of mistakes that are typically made), but I'm trying to balance the level of detail vs the length of the error message - which is already quite long.

I don't think it's necessary to mention these characters, for the same reason I don't say "if you have extra numbers on the end, take them off too".

I'd like to merge this as-is, and then I can see from the metrics in the future whether users are still getting confused or not :-)

Reply:
(The hidden characters case is much less common than all of the other mistakes, so not something we need to focus on too much; I just wanted to improve it slightly in this PR since the error not showing the character at all made things particularly confusing for users to debug)

Reply:
Yeah, this seems fine to me as-is since it's already a big improvement over the state before this PR. While I agree with Ed that the suggested "what to do" message is probably a little much - we can discuss this out of this PR and get this merged IMO.

I'll leave an approve.